docs/tutorial.md

```diff
@@ -839,3 +839,132 @@ The `fu_dump_full` and `fu_dump_raw` functions implement the
 printing of a binary buffer to the console as a stream of bytes in
 hexadecimal. See `libfwupdplugin/fu-common.c` for their definitions, you
 can find many examples of how to use them in the plugins code.
+
+## The rustgen Helper
+
+The rustgen script generates C source files that allow parsing, modifying and
+querying a packed structure or enumeration.
+This functionality is provided as parsing untrusted structured data from devices
+or firmware files is something fwupd does *a lot*, and so it makes sense to
+abstract out common code for maintainability reasons.
+It also allows us to force best-practices into the plugins without having to
+do careful review of buffer reading and writing.
+
+Structures support integers of specific widths, arrays, GUIDs, strings, default
+and constant data of variable size.
+The generated code is endian safe and if used correctly, is also safe
+against malicious data.
+
+In most cases the structure or enumeration will be defined in a `.rs`
+file -- which is the usual file extension of Rust programs.
+This was done as the format is heavily inspired by Rust, and it makes editor
+highlighting support work correctly.
+Although these files *look like* Rust files they're *not actually compiled by
+rustc*, so small differences may be noticeable.
```

Comment:
I had assumed the eventual intention was to swap out rustc for these. Is that not true?

Reply:
Eventually; but I don't want to scare anybody off with a rust dep right now.